main/modes/mainMenu/mainMenu.c

```diff
@@ -59,17 +60,48 @@ mainMenu_t* mainMenu;
 
 static const char settingsLabel[] = "Settings";
 
-static const char tftSettingLabel[]         = "TFT";
-static const char ledSettingLabel[]         = "LED";
-static const char bgmVolSettingLabel[]      = "BGM";
-static const char sfxVolSettingLabel[]      = "SFX";
-static const char micSettingLabel[]         = "MIC";
-static const char screenSaverSettingLabel[] = "Screen Saver";
+static const char tftSettingLabel[]    = "TFT";
+static const char ledSettingLabel[]    = "LED";
+static const char bgmVolSettingLabel[] = "BGM";
+static const char sfxVolSettingLabel[] = "SFX";
+static const char micSettingLabel[]    = "MIC";
+
+static const int32_t screenSaverSettingsValues[] = {
+    0,   // Off
+    10,  // 10sec
+    20,  // 20sec
+    30,  // 30sec
+    60,  // 60sec
+    120, // 2min
+    300, // 5min
+};
+
+static const char* const screenSaverSettingsOptions[] = {
+    "Screensaver: Off", "Screensaver: 10s", "Screensaver: 20s", "Screensaver: 30s",
+    "Screensaver: 1m",  "Screensaver: 2m",  "Screensaver: 5m",
+};
```

Comment:
Hrm, duplicated text. How sane is it to have a single key string, a bunch of value strings, and `strcat()` at runtime?

Reply:
Ah, yeah that's probably better. And, I think, solves the problem that checking for which label was used is really weird in the callback this way.